delensalot/biases/iterbiasesN0N1.py

```diff
@@ -0,0 +1,424 @@
+                version: wN1 includes N1 in all iterations; wE includes imperfect knowledge of E in iterations
+                recache: recomputes all instead of loading the cached arrays
+
+            Returns:
+                N0_biased: N0 with fiducial CMB Cls 
+                N1_biased_spl: N1 with fiducial CMB Cls
+                r_gg_fid: Fiducial response 
+                r_gg_true: True response, with true CMB spectra
+
+            Note: N0 and N1 output are normalized with fiducial response
+        """
+
+        (nlev_t, nlev_p, beam, lmin_ivf, lmax_ivf, lmax_qlm) = self.config
+        cls_unl_fid = self.fidcls_unl
+        cls_noise_fid = self.fidcls_noise
+
+        if cls_noise_true is None and cls_unl_true is None and fn is None:
+            fn = f'n0n1_ref_{qe_key}' + '_it' + str(itrmax)
+
+        if cls_noise_true is None: cls_noise_true = cls_noise_fid
+        if cls_unl_true is None: cls_unl_true = cls_unl_fid
+
+
+        if fn is None:
+            fn = 'n0n1_' + str(qe_key) + '_it' + str(itrmax) + '_' + _dicthash(cls_noise_true, lmax_ivf, keys=['tt', 'ee', 'bb']) + _dicthash(cls_unl_true, 6000, keys = ['tt', 'te', 'ee', 'pp'])
+            if version != '':
+                fn = 'v' + version + fn
+        if not self._cacher.is_cached(fn) or recache:
+            fid_delcls, true_delcls = self.delcls(qe_key, itrmax, cls_unl_true, cls_noise_true, version=version)
+            N0_biased, N1_biased_spl, r_gg_fid, r_gg_true = cls2N0N1(qe_key, fid_delcls[-1], true_delcls[-1],
+                                                                cls_noise_fid, cls_noise_true, lmin_ivf, lmax_ivf, lmax_qlm, doN1mat=False)
+            self._cacher.cache(fn, np.array([N0_biased, N1_biased_spl, r_gg_fid, r_gg_true]))
+            return np.array([N0_biased, N1_biased_spl, r_gg_fid, r_gg_true])
+        return self._cacher.load(fn)
+
+    def delcls(self, qe_key:str, itrmax:int, cls_unl_true: dict or None, cls_noise_true:dict or None, version=''):
+        """Returns fiducial and true partially delensed cls
+
+        """
+        (nlev_t, nlev_p, beam, lmin, lmax, lmax_qlm) = self.config
+        cls_unl_fid = self.fidcls_unl
+        cls_noise_fid = self.fidcls_noise
+
+        if cls_noise_true is None: cls_noise_true = cls_noise_fid
+        if cls_unl_true is None: cls_unl_true = cls_unl_fid
+        fid_delcls, true_delcls = get_delcls(qe_key, itrmax, cls_unl_fid, cls_unl_true, cls_noise_fid,
+                                                 cls_noise_true, lmin, lmax, lmax_qlm, version=version)
+        return fid_delcls, true_delcls
+
+
+def get_fals(qe_key, cls_cmb_filt, cls_cmb_dat, cls_noise_filt, cls_noise_dat, lmin_ivf, lmax_ivf):
+    """
+    Get the filtering Cls and from the fiducial CMB Cls and noise, as well as the IVF data Cls
+    Returns as well the QE weights and CMB response functions 
+    
+    Args:
+        qe_key: 'ptt', 'p_p', or 'p'
```

Comment:
We probably need an assert here to catch other keys this function does not support?